pxtsim/debugger.ts

```diff
@@ -124,29 +124,53 @@ namespace pxsim {
                     throw new Error();
             }
         }
-        function frameVars(frame: any) {
+        function frameVars(frame: any, fields?: string[]) {
             const r: Variables = {}
             for (let k of Object.keys(frame)) {
                 // skip members starting with __
                 if (!/^__/.test(k) && /___\d+$/.test(k)) {
                     r[k.replace(/___\d+$/, '')] = valToJSON(frame[k])
                 }
             }
-            if (frame.fields) {
+            if (frame.fields && fields) {
                 // Fields of an object.
-                for (let k of Object.keys(frame.fields)) {
+                for (let k of fields) {
+                    k = k.substring(k.lastIndexOf(".") + 1);
+                    r[k] = valToJSON(evalGetter(frame.vtable.iface[k], frame));
+                }
+            }
+            if (frame.fields) {
+                for (let k of Object.keys(frame.fields).filter(field => !field.startsWith('_'))) {
                     r[k.replace(/___\d+$/, '')] = valToJSON(frame.fields[k])
                 }
             } else if (Array.isArray(frame.data)) {
                 // This is an Array.
-               (frame.data as any[]).forEach((element, index) => {
+                (frame.data as any[]).forEach((element, index) => {
                     r[index] = valToJSON(element);
                 });
             }
             return r
         }
 
-        return frameVars(v);
+        return frameVars(v, fields);
+    }
+
+    function evalGetter(fn: LabelFn, target: RefObject) {
+        let p: any = {
```

Comment:
Let's add some comments (or maybe not use single character var names)

Reply:
(this was my fault)

Reply:
Sure!